test/secmemtest.c

```diff
@@ -75,12 +75,31 @@ static int test_sec_mem(void)
     TEST_ptr_null(OPENSSL_secure_malloc((size_t)-1));
     TEST_true(CRYPTO_secure_malloc_done());
 
+    /*-
+     * There was also a possible infinite loop when the number of
+     * elements was 1<<31, as |int i| was set to that, which is a
+     * negative number. However, it requires minimum input values:
+     *
+     * CRYPTO_secure_malloc_init((size_t)1<<34, (size_t)1<<4);
+     *
+     * Which really only works on 64-bit systems, and even then the
+     * code attempts to allocate 16 GB secure memory arena. Linux
+     * can deal with this better than other Unixy OS's (e.g. MacOS)
```

Comment:
We've seen use of several hundred MBs of locked/don't dump memory. I have not seen it fail after 64KB on Linux, it usually seems to work, although, I would have to confirm.
IMHO, the application would have to raise any limits.
In the last test, a infinite loop occurred in sh_init() before the mmap()/mprotect/mlock()/madvise(), so we really don't care if those fail, that's not the point of the test, we just don't want an infinite loop to occur.